packages/web/src/components/Stopwatch.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useState, useRef, useEffect } from "react";
+import "../Stopwatch.css";
+
+const formatTime = (time: number): string => {
+  const getSeconds = `0${time % 60}`.slice(-2);
+  const getMinutes = Math.floor(time / 60);
+  return `${getMinutes}:${getSeconds}`;
+};
+
+const Stopwatch: React.FC = () => {
+  const [isRunning, setIsRunning] = useState(false);
+  const [isEnabled, setIsEnabled] = useState(true);
+  const [remainingTime, setRemainingTime] = useState(0);
+  const [timeInput, setTimeInput] = useState(0);
+  const [hasStarted, setHasStarted] = useState(false);
+  const [savedRemainingTime, setSavedRemainingTime] = useState(0);
+  const [savedTimeInput, setSavedTimeInput] = useState(0);
+  const [displayedProgress, setDisplayedProgress] = useState(0);
+  const [isTransitioning, setIsTransitioning] = useState(false);
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  const circumference = 282.743;
+  const progress =
+    timeInput > 0 ? (remainingTime / timeInput) * circumference : 0;
+  const dashOffset = circumference - displayedProgress;
+
+  useEffect(() => {
```

Comment:
The logic in this useEffect is a bit confusing even if probably correct. You start the timer when `isRunning` and there is `remainingTime` which in my mind means that it should be running already. You could consider splitting the logic in 2 useEffect, one that triggers the timer (or turns it off) and another that update the state.

Reply:
Well, you actually don't need a useEffect to start your timer, you could just create it in `startCountdown` (or from a function invoked there).

Reply:
Good suggestions. I incorporated them into commit c084a7702ea9d4af7cc3fef1688f404632c8d89f "refactor stopwatch and make the logic clearer"